src/azure-cli/azure/cli/command_modules/synapse/operations/accesscontrol.py

```diff
@@ -4,55 +4,67 @@
+from .._client_factory import cf_synapse_role_assignments, cf_synapse_role_definitions, cf_graph_client_factory
+from ..constant import ITEM_NAME_MAPPING
 
 
 # List Synapse Role Assignment
-def list_role_assignments(cmd, workspace_name, role=None, assignee=None):
-    # get role id
-    role_id = _resolve_role_id(cmd, role, workspace_name)
-    # get object_id
-    object_id = _resolve_object_id(cmd, assignee, fallback_to_object_id=True)
+def list_role_assignments(cmd, workspace_name, role=None, assignee=None, assignee_object_id=None,
+                          scope=None, item=None, item_type=None):
+    if bool(assignee) and bool(assignee_object_id):
+        raise CLIError('usage error: --assignee STRING | --assignee-object-id GUID')
+
+    if bool(item) != bool(item_type):
+        raise CLIError('usage error: --item-type STRING --item STRING')
+
+    return _list_role_assignment(cmd, workspace_name, role, assignee or assignee_object_id,
+                                 scope, resolve_assignee=(not assignee_object_id), item=item, item_type=item_type)
 
-    client = cf_synapse_client_accesscontrol_factory(cmd.cli_ctx, workspace_name)
-    role_assignments = client.get_role_assignments(role_id, object_id)
 
-    # TODO:
-    # Currently, when only `ObjectId` is specified, the cmdlet returns incorrect result.
-    # Filter from client side as a workaround
-    if object_id:
-        role_assignments = [x for x in role_assignments if x.principal_id == object_id]
+def _list_role_assignment(cmd, workspace_name, role=None, assignee=None, scope=None,
+                          resolve_assignee=True, item=None, item_type=None):
+    """Prepare scope, role ID and resolve object ID from Graph API."""
+    if any([scope, item, item_type]):
+        scope = _build_role_scope(workspace_name, scope, item, item_type)
+    role_id = _resolve_role_id(cmd, role, workspace_name)
+    object_id = _resolve_object_id(cmd, assignee, fallback_to_object_id=True) if resolve_assignee else assignee
+    client = cf_synapse_role_assignments(cmd.cli_ctx, workspace_name)
+    role_assignments = client.list_role_assignments(role_id, object_id, scope).value
+    if scope:
+        role_assignments = [x for x in role_assignments if x.scope == scope]
     return role_assignments
 
 
 # Show Synapse Role Assignment By Id
 def get_role_assignment_by_id(cmd, workspace_name, role_assignment_id):
-    client = cf_synapse_client_accesscontrol_factory(cmd.cli_ctx, workspace_name)
+    client = cf_synapse_role_assignments(cmd.cli_ctx, workspace_name)
     return client.get_role_assignment_by_id(role_assignment_id)
 
 
 # Delete Synapse Role Assignment
-def delete_role_assignment(cmd, workspace_name, ids=None, assignee=None, role=None):
-    client = cf_synapse_client_accesscontrol_factory(cmd.cli_ctx, workspace_name)
+def delete_role_assignment(cmd, workspace_name, ids=None, assignee=None, assignee_object_id=None, role=None,
```

Comment:
delete_role_assignment_by_id?

Reply:
Cx can give a set of ids to delete not only one role assignment. Besides, user can delete role assignments by assignee or by role. 

delete_role_assignment_by_id is not friendly to cx.